openvdb_ax/openvdb_ax/compiler/PointExecutable.h

```diff
@@ -0,0 +1,148 @@
+// Copyright Contributors to the OpenVDB Project
+// SPDX-License-Identifier: MPL-2.0
+
+/// @file compiler/PointExecutable.h
+///
+/// @authors Nick Avramoussis, Francisco Gochez, Richard Jones
+///
+/// @brief The PointExecutable, produced by the OpenVDB AX Compiler for
+///   execution over OpenVDB Points Grids.
+///
+
+#ifndef OPENVDB_AX_COMPILER_POINT_EXECUTABLE_HAS_BEEN_INCLUDED
+#define OPENVDB_AX_COMPILER_POINT_EXECUTABLE_HAS_BEEN_INCLUDED
+
+#include "../compiler/CustomData.h"
+#include "../compiler/AttributeRegistry.h"
+
+#include <openvdb/openvdb.h>
+#include <openvdb/points/PointDataGrid.h>
+
+#include <unordered_map>
+
+class TestPointExecutable;
+
+namespace llvm {
+class ExecutionEngine;
+class LLVMContext;
+}
+
+namespace openvdb {
+OPENVDB_USE_VERSION_NAMESPACE
+namespace OPENVDB_VERSION_NAME {
+namespace ax {
+
+class Compiler;
+
+/// @brief Object that encapsulates compiled AX code which can be executed on a target point grid
+class PointExecutable
+{
+public:
+    using Ptr = std::shared_ptr<PointExecutable>;
+    ~PointExecutable();
```

Comment:
Empty deconstructors should just be `= default` or made implicit.

Reply:
This can't be defaulted here because there definition of `Settings` lives in the .cc. If we change this class then perhaps this can be done